### PR TITLE
fix(config-map): update examples in documentation

### PR DESCRIPTION
**Pull request by vinayak-kukreja**
Currently these examples are not working if a user just copy and pastes them to try them out. 
These even have some properties missing which throws syntax errors.

Signed-off-by: Vinayak Kukreja <vinakuk@amazon.com>

### CODE REVIEW

File: docs/plus/config-map.md [hunk 1]
```diff
@@ -42,29 +62,37 @@ Here is a nifty little trick you can use to create a volume that contains a dire
...
+      workingDir: mountPath,
+    });
+
+    // from here, just mount the volume to a container, and run your app!
+    container.mount(mountPath, appVolume);
+  }
+}
+
+const app = new App();
+new MyChart(app, 'config-map');
```
Comment (ryparker) [inline docs/plus/config-map.md]:
```suggestion
new MyChart(app, 'AppWithDir');
```

What do you think about using a more descriptive name like this?

File: docs/plus/config-map.md [hunk 2]
```diff
@@ -42,29 +62,37 @@ Here is a nifty little trick you can use to create a volume that contains a dire
...
-  workingDir: mountPath,
-});
-
-// from here, just mount the volume to a container, and run your app!
-container.mount(mountPath, appVolume);
+import { Construct } from 'constructs';
+import { App, Chart, ChartProps } from 'cdk8s';
+
+export class MyChart extends Chart {
+  constructor(scope: Construct, id: string, props: ChartProps = { }) {
```
Comment (ryparker) [inline docs/plus/config-map.md]:
```suggestion
  constructor(scope: Construct, id: string, props?: ChartProps) {
```

⛏️ props aren't required on Chart.

File: docs/plus/config-map.md [hunk 3]
```diff
@@ -42,29 +62,37 @@ Here is a nifty little trick you can use to create a volume that contains a dire
...
+    super(scope, id, props);
+    const appMap = new kplus.ConfigMap(this, 'Config');
+
+    // add the files in the directory to the config map.
+    // this will create a key for each file.
+    // note: this directory needs to exist
+    // note: that only top level files will be included, sub-directories are not yet supported.
+    appMap.addDirectory(path.join(__dirname, 'app'));
+
+    const appVolume = kplus.Volume.fromConfigMap(this, 'config-map', appMap);
```
Comment (ryparker) [inline docs/plus/config-map.md]:
```suggestion
    const appVolume = kplus.Volume.fromConfigMap(this, 'ConfigMap', appMap);
```

⛏️ Pascal case ids for consistency.

File: docs/plus/config-map.md [hunk 4]
```diff
@@ -27,13 +39,21 @@ You can create config maps and add some data to them like so:
...
+  constructor(scope: Construct, id: string, props: ChartProps = { }) {
+    super(scope, id, props);
+    
+    const config = new kplus.ConfigMap(this, 'Config');
+    config.addData('url', 'https://my-endpoint:8080');
+  }
+}
+
+const app = new App();
+new MyChart(app, 'config-map');
```
Comment (ryparker) [inline docs/plus/config-map.md]:
```suggestion
new MyChart(app, 'ConfigMap');
```

File: docs/plus/config-map.md [hunk 5]
```diff
@@ -13,12 +13,24 @@ and will therefore not be included in the resulting manifest.
...
 
-const config: kplus.IConfigMap = kplus.ConfigMap.fromConfigMapName('config');
+export class MyChart extends Chart {
+  constructor(scope: Construct, id: string, props: ChartProps = { }) {
+    super(scope, id, props);
 
-// the 'config' constant can later be used by API's that require an IConfigMap.
-// for example when creating a volume.
-const volume = kplus.Volume.fromConfigMap(config);
+    const config: kplus.IConfigMap = kplus.ConfigMap.fromConfigMapName(this, 'config-map', 'config');
```
Comment (ryparker) [inline docs/plus/config-map.md]:
```suggestion
    const config: kplus.IConfigMap = kplus.ConfigMap.fromConfigMapName(this, 'ConfigMap', 'config');
```

⛏️  Pascal case ids for consistency.

File: docs/plus/config-map.md [hunk 6]
```diff
@@ -13,12 +13,24 @@ and will therefore not be included in the resulting manifest.
...
+
+    // the 'config' constant can later be used by API's that require an IConfigMap.
+    // for example when creating a volume.
+    kplus.Volume.fromConfigMap(this, 'volume', config);
```
Comment (ryparker) [inline docs/plus/config-map.md]:
```suggestion
    kplus.Volume.fromConfigMap(this, 'Volume', config);
```

File: docs/plus/config-map.md [hunk 7]
```diff
@@ -13,12 +13,24 @@ and will therefore not be included in the resulting manifest.
...
+  }
+}
+
+const app = new App();
+new MyChart(app, 'config-map');
```
Comment (ryparker) [inline docs/plus/config-map.md]:
```suggestion
new MyChart(app, 'VolumeFromConfigMap');
```

What do you think about using a more descriptive name like this?

File: docs/plus/config-map.md [hunk 8]
```diff
@@ -27,13 +39,21 @@ You can create config maps and add some data to them like so:
...
-const app = new k.App();
-const chart = new k.Chart(app, 'Chart');
-
-const config = new kplus.ConfigMap(chart, 'Config');
-config.addData('url', 'https://my-endpoint:8080');
+import { Construct } from 'constructs';
+import { App, Chart, ChartProps } from 'cdk8s';
+
+export class MyChart extends Chart {
+  constructor(scope: Construct, id: string, props: ChartProps = { }) {
```
Comment (ryparker) [inline docs/plus/config-map.md]:
```suggestion
  constructor(scope: Construct, id: string, props?: ChartProps) {
```

⛏️ `props` aren't required on `Chart`.

**Review by ryparker — CHANGES_REQUESTED** [review]

**Comment by ryparker** [comment]
Good stuff. Mostly just nitpick ⛏️  suggestions. Shouldn't block the PR though.

File: docs/plus/config-map.md (hunk 1 above)
Comment (vinayak-kukreja) [reply docs/plus/config-map.md]: That's better. Will make the update.

File: docs/plus/config-map.md (hunk 3 above)
Comment (vinayak-kukreja) [reply docs/plus/config-map.md]: I see, will remember this. Thank you :)

File: docs/plus/config-map.md (hunk 7 above)
Comment (vinayak-kukreja) [reply docs/plus/config-map.md]: That's better. Will make the update.

**Review by ryparker — APPROVED** [review]
💯

**Comment by vinayak-kukreja** [comment]
## 💔 Backport failed
The pull request could not be backported due to the following error:
`Missing target branch choices`

### Manual backport
To create the backport manually run:
```
backport --pr 1334
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by vinayak-kukreja** [comment]
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|k8s-23/main,|The branch "k8s-23/main," does not exist|
|✅|k8s-24/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/1337)|

### Manual backport
To create the backport manually run:
```
backport --pr 1334
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by vinayak-kukreja** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-23/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/1338)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**